tails_files/install.sh

```diff
@@ -68,23 +87,360 @@ if [ -f $XSESSION_RC ]; then
+  chmod 600 "${securedrop_dotfiles}/securedrop_icon.png"
+}
+
+function prompt_for_document_aths_info()
+{
+  # Ad-hoc memoization via a global variable. If we've already prompted for
+  # the ATHS info here, the global will be populated. Otherwise, we'll store it.
+  if [ -z $document_aths_info_global ] ; then
+    # During initial provisioning of the Journalist Workstation, the ATHS file
+    # for the Document Interface may not be present. If not, prompt for the info.
+    aths_regex="^(HidServAuth [a-z2-7]{16}\.onion [A-Za-z0-9+/.]{22})"
+    # Loop while reading the input from a dialog box.
+    while [[ ! "$document_aths_info" =~ $aths_regex ]]; do
+      document_aths_info="$(zenity --entry \
+        --title='Hidden service authentication setup' \
+        --width=600 \
+        --window-icon=${securedrop_dotfiles}/securedrop_icon.png \
+        --text='Enter the HidServAuth value to be added to /etc/tor/torrc:')"
+    done
+  # The global var is populated, so use it for the function var.
+  else
+    document_aths_info="$document_aths_info_global"
+  fi
+  echo "$document_aths_info"
+}
+
+function prompt_for_source_ths_url()
+{
+  # Interactively prompt for the Source Interface Onion URL.
+  # During initial provisioning of the Journalist Workstation, the THS file
+  # for the Source Interface may not be present. If not, prompt for the info.
+
+  # Ad-hoc memoization via a global variable. If we've already prompted for
+  # the ATHS info here, the global will be populated. Otherwise, we'll store it.
+  if [ -z $source_ths_url_global ] ; then
+    ths_regex="^[a-z2-7]{16}\.onion"
+    # Loop while reading the input from a dialog box.
+    while [[ ! "$document_aths_info" =~ $ths_regex ]]; do
+      ths_url="$(zenity --entry \
+        --title='Hidden service authentication setup' \
+        --width=600 \
+        --window-icon=${securedrop_dotfiles}/securedrop_icon.png \
+        --text='Enter the Source Interface Onion URL:')"
+    done
+    # Store value in script global for reuse without prompting.
+    source_ths_url_global="$ths_url"
+
+  # The global var is populated, so use it for the function var.
+  else
+    ths_url="$source_ths_url_global"
+  fi
+  echo "$ths_url"
+}
+
+function lookup_app_ssh_aths_url()
+{
+  app_ssh_aths="$(awk '{ print $2 }' ${securedrop_ansible_base}/app-ssh-aths)"
+  echo "$app_ssh_aths"
+}
+
+function lookup_mon_ssh_aths_url()
+{
+  mon_ssh_aths="$(awk '{ print $2 }' ${securedrop_ansible_base}/mon-ssh-aths)"
+  echo "$mon_ssh_aths"
+}
+
+function lookup_document_aths_url()
+{
+  # First look for the flat file containing the exact value we want.
+  # The Admin Workstation will certainly have this file, and the Journalist
+  # Workstation probably, but not definitely.
+  if [ -f "${securedrop_ansible_base}/app-document-aths" ] ; then
+    app_document_aths="$(awk '{ print $2 }' ${securedrop_ansible_base}/app-document-aths)"
+  # Failing that, check for ATHS info in the existing Tor config. A previously
```

Comment:
Thinking ahead, we should probably support provisioning the same JW to be used for new ATHSes, i.e. rotated hidden service private keys and URLs. This happened during [Heartbleed](https://securedrop.org/news/securedrop-and-openssl-vulnerability) and it could be necessary again with next-generation Hidden Services (elliptic curve), or server compromise. However, if rotating keys and URLs is not currently supported in the SecureDrop server playbooks, then it shouldn't be in tails_files.